### PR TITLE
Fix session termination issue

### PR DESCRIPTION
**Pull request by piraveena**
### Proposed changes in this pull request

Resolves https://github.com/wso2-enterprise/asgardeo-product/issues/2181

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java [hunk 1]
```diff
@@ -107,6 +107,9 @@ public AuthorizationGrantCacheEntry getValueFromCacheByTokenId(AuthorizationGran
     public AuthorizationGrantCacheEntry getValueFromCacheByToken(AuthorizationGrantCacheKey key) {
         AuthorizationGrantCacheEntry cacheEntry = super.getValueFromCache(key);
         if (cacheEntry == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Getting cache entry from session store using access token");
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java:111]: Put some context identifier to identify the log.  Ex. key.getUserAttributesId(

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java [hunk 2]
```diff
@@ -155,6 +158,9 @@ public void addToCacheByCode(AuthorizationGrantCacheKey key, AuthorizationGrantC
     public AuthorizationGrantCacheEntry getValueFromCacheByCode(AuthorizationGrantCacheKey key) {
         AuthorizationGrantCacheEntry cacheEntry = super.getValueFromCache(key);
         if (cacheEntry == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Getting cache entry from session store using authorization code");
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java:162]: Put some context identifier to identify the log. Ex. key.getUserAttributesId(

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java (hunk 1 above)
Comment (piraveena) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java:111]: That AuthorizationGrantCacheKey has either authorization code/ token. That is why I didn't log them

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java (hunk 2 above)
Comment (piraveena) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java:162]: That AuthorizationGrantCacheKey has either authorization code/ token. That is why I didn't log them

Comment (piraveena) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/cache/AuthorizationGrantCache.java]: added two logs. if tokenis loggable, then logged the hash of the accesstoken/code. Else logged only the log string

**Review by IsuraD — APPROVED** [review]

**Pull request merged**